### PR TITLE
NAS-4284 documentation update

### PR DESCRIPTION
**Pull request by hkad98**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by jaceksan — CHANGES_REQUESTED** [review]

File: gooddata-sdk/docs/services/catalog-workspace-content.rst [hunk 1]
```diff
@@ -31,6 +27,42 @@ The *gooddata_sdk.catalog_workspace_content* supports the following entity API c
 
     Retrieve all datasets with attributes, facts, and metrics for a workspace.
 
+* ``get_attributes_catalog(workspace_id: str)``
+
+    Returns *list[CatalogAttribute]*
+
+    # TODO
```
[the remark this answers is not in the document]
Comment (hkad98) [reply gooddata-sdk/docs/services/catalog-workspace-content.rst:34]: Nope. I forgot. I will fix it. Thanks.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/docs/services/catalog-data-source.rst [hunk 2]
```diff
@@ -209,6 +213,35 @@ The *gooddata_sdk.catalog_data_source* supports the following declarative API ca
...
+
+    Returns *CatalogDeclarativeTables*.
+
+    Retrieve physical model for a given data source.
+
+* ``put_declarative_pdm(data_source_id: str, declarative_tables: CatalogDeclarativeTables)``
+
+    Set physical model for a given data source.
+
+* ``store_declarative_pdm(data_source_id: str, layout_root_path: Path = Path.cwd())``
```
[the remark this answers is not in the document]
Comment (hkad98) [reply gooddata-sdk/docs/services/catalog-data-source.rst:229]: Fixed.

File: gooddata-sdk/docs/services/catalog-data-source.rst [hunk 3]
```diff
@@ -209,6 +213,35 @@ The *gooddata_sdk.catalog_data_source* supports the following declarative API ca
...
+    Returns *CatalogDeclarativeTables*.
+
+    Load declarative physical model layout, which was stored using *store_declarative_pdm* for a given data source.
+
+* ``load_and_put_declarative_pdm(self, data_source_id: str, layout_root_path: Path = Path.cwd())``
+
+    This method combines *load_declarative_pdm* and
+    *put_declarative_pdm* methods to load and set
+    layouts stored using *store_declarative_pdm*.
+
```
[the remark this answers is not in the document]
Comment (hkad98) [reply gooddata-sdk/docs/services/catalog-data-source.rst:254]: Added.

File: gooddata-sdk/docs/services/catalog-data-source.rst [hunk 4]
```diff
@@ -266,6 +298,11 @@ The *gooddata_sdk.catalog_data_source* supports the following action API calls:
 
     Returns a list of schemas that exist in the database and can be configured in the data source entity. Data source managers like Dremio or Drill can work with multiple schemas and schema names can be injected into scan_request to filter out tables stored in the different schemas.
 
+* ``test_data_sources_connection(declarative_data_sources: CatalogDeclarativeDataSources, credentials_path: Optional[Path] = None)``
+
+    Tests connection to declarative data sources. We can specify an optional path to credentials.
```
[the remark this answers is not in the document]
Comment (hkad98) [reply gooddata-sdk/docs/services/catalog-data-source.rst]: Done.

**Review by jaceksan — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR141

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/127/ : SUCCESS in 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/127/ : FAILURE in 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/129/ : SUCCESS in 20s

**Pull request merged**